xarray/coding/times.py

```diff
@@ -203,18 +238,50 @@ def _infer_time_units_from_diff(unique_timedeltas):
     return 'seconds'
 
 
+def infer_calendar_name(dates):
+    """Given an array of datetimes, infer the CF calendar name"""
+    if np.asarray(dates).dtype == 'datetime64[ns]':
+        return 'proleptic_gregorian'
+    else:
+        try:
+            return np.asarray(dates)[0].calendar
+        except IndexError:
+            return np.asarray(dates).item().calendar
```

Comment:
Use something like `np.asarray(dates).ravel()[0]` to avoid the need for try/except.